cmd/resolutioncli/entity_source.go

```diff
@@ -0,0 +1,154 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/model"
+	"github.com/operator-framework/operator-registry/alpha/property"
+)
+
+type indexRefEntitySource struct {
+	renderer      action.Render
+	entitiesCache input.EntityList
+}
+
+func NewIndexRefEntitySourceEntitySource(indexRef string) *indexRefEntitySource {
+	return &indexRefEntitySource{
+		renderer: action.Render{
+			Refs:           []string{indexRef},
+			AllowedRefMask: action.RefDCImage | action.RefDCDir,
+		},
+	}
+}
+
+func (es *indexRefEntitySource) Get(ctx context.Context, id deppy.Identifier) (*input.Entity, error) {
+	panic("not implemented")
+}
+
+func (es *indexRefEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+	entities, err := es.entities(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resultSet := input.EntityList{}
+	for _, entity := range entities {
+		if filter(&entity) {
+			resultSet = append(resultSet, entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es *indexRefEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+	entities, err := es.entities(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resultSet := input.EntityListMap{}
+	for _, entity := range entities {
+		keys := fn(&entity)
+		for _, key := range keys {
+			resultSet[key] = append(resultSet[key], entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es *indexRefEntitySource) Iterate(ctx context.Context, fn input.IteratorFunction) error {
+	entities, err := es.entities(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if err := fn(&entity); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (es *indexRefEntitySource) entities(ctx context.Context) (input.EntityList, error) {
+	if es.entitiesCache == nil {
+		cfg, err := es.renderer.Run(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		model, err := declcfg.ConvertToModel(*cfg)
+		if err != nil {
+			return nil, err
+		}
+
+		entities, err := modelToEntities(model)
+		if err != nil {
+			return nil, err
+		}
+
+		es.entitiesCache = entities
+	}
+
+	return es.entitiesCache, nil
+}
+
+func modelToEntities(model model.Model) (input.EntityList, error) {
```

Comment:
This somehow needs to share what we have in here:

https://github.com/operator-framework/operator-controller/blob/098778ea18bd1ea501795b1a68c5e52dbdff2515/internal/resolution/entitysources/catalogdsource.go#L75-L120

So we don't have to maintain two translations in parallel.